src/components/form/SegmentRadioField/SegmentRadioField.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+import { AppTheme } from '~/models/AppTheme ';
+import Colors from '~/theme/Colors';
+
+import {
+  StyledText,
+  StyledTouchableOpacity,
+  StyledView,
+} from './SegmentRadioField.styled';
+
+export interface SegmentRadioFieldProps {
+  touched?: boolean;
+  theme?: AppTheme;
+  value: any;
+  onPress(value: any): void;
```

Comment:
And how to use it? :D You meant to make generic component? :thinking:

Reply:
Okay, I thought it would be inferred right away
```
const someProps: SegmentRadioFieldProps = {
  value: 5,
  // val inferred as number
  onPress: (val) => {}
}
```
but it's not inferred. It looks like you need to make the component generic. Then value seems to be correctly inferred :smile:  Do you need an example?